sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpRetryPolicy.java

```diff
@@ -310,7 +311,14 @@ private void logRetry(LoggingEvent log, int tryCount, Duration delayDuration, Ru
         }
     }
 
-    private Duration calculateRetryDelay(int retryAttempts) {
+    private Duration calculateRetryDelay(HttpRetryCondition retryCondition) {
+        if (delayFromRetryCondition != null) {
+            Duration delay = delayFromRetryCondition.apply(retryCondition);
+            if (delay != null) {
+                return delay;
+            }
```

Comment:
This seems like an invalid `delayFromRetryCondition` function if it returns `null` and should throw instead of falling back to other types delay durations.

Reply:
Good catch, will change this so if that returns null it falls back to the other handling